### DISCUSSION TITLE
Forward differentiation nan gradient but reverse mode is fine

### DISCUSSION THREAD

**Discussion by unalmis** [jax-ml/jax#25657] [Q&A]
Under what circumstances is it possible to have a forward mode nan gradient but not a reverse mode nan gradient? The context for this issue is [here](https://github.com/PlasmaControl/DESC/issues/1489).

**Comment by bmaxdk** [comment]
If the very first gradient evaluation in forward-mode hits a near-zero denominator, you get a NaN right away. Can you  use a small safe divisor offset (ϵ) or clamping? Since extremely small value can cause this issue
